### PR TITLE
Fix missing ⬇️ event on editable ComboBox

### PR DESCRIPTION
**Pull request by hamen**
To keep the level of noise manageable, I suggest a commit-by-commit review.

### CODE REVIEW

**Comment by hamen** [comment]
I had to run this #46a9b67 on Mac to satisfy the linter. This is so annoying.

File: ui-tests/src/test/kotlin/ListComboBoxUiTest.kt [hunk 1]
```diff
@@ -239,14 +240,12 @@ class ListComboBoxUiTest {
...
+        val comboBox = disabledEditableComboBox()
         comboBox.assertIsDisplayed().assertHasNoClickAction().performClick()
-        composeRule
-            .onNodeWithTag("Jewel.ComboBox.NonEditableText")
-            .assertIsDisplayed()
-            .assertHasNoClickAction()
-            .performClick()
+
+        // Ivan: It would be nice to check the absence of the OnClick,
+        // but I believe the BasicTextField adds it even on a disable state 🤷
```
Comment (rock3r) [inline ui-tests/src/test/kotlin/ListComboBoxUiTest.kt:247]: >I believe the BasicTextField adds it even on a disable state

If this is true and the clickable is not enabled = false, that is a problem and it needs to be reported

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt [hunk 2]
```diff
@@ -166,30 +166,30 @@ public fun EditableComboBox(
...
 
                 Chevron(
                     isEnabled = isEnabled,
                     style = style,
                     interactionSource = interactionSource,
-                    onHoveredChanged = { chevronHovered = it },
+                    onHoveredChange = { chevronHovered = it },
                     setPopupExpanded = { popupExpanded = it },
-                    onPressWhenEnabled = onPressWhenEnabled,
+                    onPressWhenEnable = onPressWhenEnabled,
```
Comment (rock3r) [inline ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt:192]: I think this change was... a bit over-eager :D `onPressWhenEnabled` is correct. I can also take `onPress` (the _when enabled_ bit can be omitted)

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt [hunk 3]
```diff
@@ -217,58 +217,59 @@ public fun EditableComboBox(
...
     popupExpanded: Boolean,
     textStyle: TextStyle,
     textFieldInteractionSource: MutableInteractionSource,
     onArrowDownPress: () -> Unit,
     onArrowUpPress: () -> Unit,
     onEnterPress: () -> Unit,
-    onSetPopupExpanded: (Boolean) -> Unit,
-    onFocusedChanged: (Boolean) -> Unit,
-    onHoveredChanged: (Boolean) -> Unit,
+    onSetPopupExpandedEvent: (Boolean) -> Unit,
```
Comment (rock3r) [inline ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt:231]: Why `event`? It's not a common naming pattern (e.g., you don't call other lambdas `onFocusedChangeEvent` or `onClickEvent`)

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt [hunk 4]
```diff
@@ -217,58 +217,59 @@ public fun EditableComboBox(
 
 @Composable
 private fun TextInput(
-    modifier: Modifier,
```
Comment (rock3r) [inline ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt:220]: This was up there and without a default value on purpose...

**Review by rock3r — COMMENTED** [review]
Apart from a bunch of unrelated/unneeded renames that should be undone, I don't understand where the "fix" actually is — it feels like a lot of non-functional changes

File: ui-tests/src/test/kotlin/ListComboBoxUiTest.kt (hunk 1 above)
Comment (hamen) [reply ui-tests/src/test/kotlin/ListComboBoxUiTest.kt:247]: I want to be sure we are holding it correctly, and if we need, I'll open a ticket to the interested team ⌛

**Pull request merged**